src/Admin/AbstractAdmin.php

```diff
@@ -692,6 +694,11 @@ public function delete($object)
         }
     }
 
+    public function onPostSubmit(FormEvent $event)
```

Comment:
Please do not add new methods to the `AbstractAdmin` class, we already have enough SRP violations on this class.

You could use a closure like this instead:
```php
->addEventListener(FormEvents::POST_SUBMIT, function (FormEvent $event) {
    $this->preValidate($event->getData());
}, 100)

Reply:
ok i did it